### ISSUE TITLE
About Train

### ISSUE THREAD

**Issue by jyx1244204317** [ge-xing/Diff-UNet#19]
def training_step(self, batch):
        image, label = self.get_input(batch)
        x_start = label

        x_start = (x_start) * 2 - 1
        x_t, t, noise = self.model(x=x_start, pred_type="q_sample")
        pred_xstart, pred_y = self.model(x=x_t, step=t, image=image, pred_type="denoise")

        loss_dice = self.dice_loss(pred_xstart, label)
        loss_bce = self.bce(pred_xstart, label)

        pred_xstart = torch.sigmoid(pred_xstart)
        loss_mse = self.mse(pred_xstart, label)

        loss = loss_dice + loss_bce + loss_mse

In the training phase, the noise generated from q_sample, the next step of prediction is the loss calculated with label. In my understanding, xt should calculate the loss with xt-1, and then generate x0 for the final target detection. But here why not calculate the loss with the next step, but directly calculate with the standard label?

**Comment by 920232796** [comment]
For the segmentation task, diffusion model need to predict x_0, rather than noise.

**Comment by jyx1244204317** [comment]
Thank you!
